src/main/java/com/streamr/client/subs/BasicSubscription.java

```diff
@@ -119,6 +120,7 @@ private synchronized void cancelGroupKeyRequest(String publisherId) {
             timer.cancel();
             timer.purge();
             pendingGroupKeyRequests.remove(publisherId);
+            getLogger().trace("Pending group key request canceled for publisher {}", publisherId);
```

Comment:
IMO put log statements right at the top so it's clearer what was happening if an error occurs?